### DISCUSSION TITLE
How to send Twillio SMS as an Automation ?

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#6593] [Help] [answered]
Any help in creating such an automation ? Is this currently possible via JS scripting or Bash scripting ?

I am trying to use the given Twillio bash command, but when i do i get the error "curl not found".
I am assuming BB is trying to execute the bash inside the docker container ( i am on self hosted / docker / v.218 )
is curl available to BB inside the container ?

**Comment by mike12345567** [comment, marked as answer]
Hi @poirazis - yes you're correct that the bash functionality executes inside the container rather than on the host, currently it is a very limited image, this might be something we change in the near future to make the bash functions more usable.

The best option you have is to use the `External Data connector` action as such:
![image](https://user-images.githubusercontent.com/4407001/179255619-761926dd-d83d-4034-98dd-abe7ab808f93.png)

You can then use our REST API interface to configure a full query as you need to the Twilio API, and simply trigger it as part of the automation flow. In my example I have configured a REST query which has a body, from and to binding, but in your case you can configure it in whatever way you need, just pass the parameters from the automation you require to the query. This allows you the most flexibility in terms of carrying out API queries from within Budibase.

**Reply by poirazis** [reply]
Thank you @mike12345567 !
